lib/rules/amd-function-arity.js

```diff
@@ -30,10 +31,41 @@ module.exports = function (context) {
         return node.type === "FunctionExpression";
     }
 
+    function findAllowedExtraDependencies() {
+        if (allowExtraDependencies) {
+            return false;
+        }
+
+        if (context.options && context.options[0] && context.options[0].allowExtraDependencies) {
```

Comment:
At this point, rather than all the duplication, I think we can wrap the property access for `allowExtraDependencies` into a function. _OR_ fetch the raw value once on line 22, be it `true`, `false`, some `Array` or `undefined`, and then worry about the defaults, etc here.